# Изучение развития игровой индустрии 

- Автор:Васильев А.В.
- Дата:23.06.2025

### Цели и задачи проекта

Целью проекта является привлечение новой аудитории игроков,а так же подготовка статьи о истории развития игр в XXI веке.Основная задача статьи рассмотреть и изучить игровые платформы,объёмы продаж игр разных жанров и региональные предпочтения игроков.В статье мы изучим развитие игровой индустрии с 2000 по 2013 год.В этот период было большое развитие игровой индустрии,появлялись новые мощьные игровые консоли,жанры видео игр,а так же расло количество многопользовательских онлайн игр! Анализ будет проводиться с историческими данными, собранными из открытых источников.В них содержиться необходимая информация о продажах игр,а также об их оценках. </font>

### Описание данных

**Данные /datasets/new_games.csv содержат информацию о продажах игр разных жанров и платформ, а также пользовательские и экспертные оценки игр:**


* Name — название игры.


* Platform — название платформы.


* Year of Release — год выпуска игры.


* Genre — жанр игры.


* NA sales — продажи в Северной Америке (в миллионах проданных копий).


* EU sales — продажи в Европе (в миллионах проданных копий).


* JP sales — продажи в Японии (в миллионах проданных копий).


* Other sales — продажи в других странах (в миллионах проданных копий).


* Critic Score — оценка критиков (от 0 до 100).


* User Score — оценка пользователей (от 0 до 10).


* Rating — рейтинг организации ESRB (англ. Entertainment Software Rating Board).</font>

### Содержимое проекта

* [Загрузка и знакомство с данными](#info)


* [Проверка ошибок в данных и их предобработка](#pr)


* [Фильтрация данных](#fd)


* [Категоризация данных](#cd)


* [Вывод по результатам анализа](#end)
</font>

## 1. Загрузка данных и знакомство с ними<a id='intro'></a>

- Загрузите необходимые библиотеки Python и данные датасета `/datasets/new_games.csv`.
- Познакомьтесь с данными: выведите первые строки и результат метода `info()`.


In [1]:
import pandas as pd
df = pd.read_csv('/datasets/new_games.csv')
display(df.info())
display(df)


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16956 entries, 0 to 16955
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Name             16954 non-null  object 
 1   Platform         16956 non-null  object 
 2   Year of Release  16681 non-null  float64
 3   Genre            16954 non-null  object 
 4   NA sales         16956 non-null  float64
 5   EU sales         16956 non-null  object 
 6   JP sales         16956 non-null  object 
 7   Other sales      16956 non-null  float64
 8   Critic Score     8242 non-null   float64
 9   User Score       10152 non-null  object 
 10  Rating           10085 non-null  object 
dtypes: float64(4), object(7)
memory usage: 1.4+ MB


None

Name Platform  Year of Release         Genre  \
0                         Wii Sports      Wii           2006.0        Sports   
1                  Super Mario Bros.      NES           1985.0      Platform   
2                     Mario Kart Wii      Wii           2008.0        Racing   
3                  Wii Sports Resort      Wii           2009.0        Sports   
4           Pokemon Red/Pokemon Blue       GB           1996.0  Role-Playing   
...                              ...      ...              ...           ...   
16951  Samurai Warriors: Sanada Maru      PS3           2016.0        Action   
16952               LMA Manager 2007     X360           2006.0        Sports   
16953        Haitaka no Psychedelica      PSV           2016.0     Adventure   
16954               Spirits & Spells      GBA           2003.0      Platform   
16955            Winning Post 8 2016      PSV           2016.0    Simulation   

       NA sales EU sales JP sales  Other sales  Critic Score User Score Rating  
0         41.36    28.96     3.77         8.45          76.0          8      E  
1         29.08     3.58     6.81         0.77           NaN        NaN    NaN  
2         15.68    12.76     3.79         3.29          82.0        8.3      E  
3         15.61    10.93     3.28         2.95          80.0          8      E  
4         11.27     8.89    10.22         1.00           NaN        NaN    NaN  
...         ...      ...      ...          ...           ...        ...    ...  
16951      0.00      0.0     0.01         0.00           NaN        NaN    NaN  
16952      0.00     0.01      0.0         0.00           NaN        NaN    NaN  
16953      0.00      0.0     0.01         0.00           NaN        NaN    NaN  
16954      0.01      0.0      0.0         0.00           NaN        NaN    NaN  
16955      0.00      0.0     0.01         0.00           NaN        NaN    NaN  

[16956 rows x 11 columns]

- Сделайте вывод о полученных данных: данные какого объёма вам предоставили, соответствуют ли они описанию, встречаются ли в них пропуски, используются ли верные типы данных.
- Отметьте другие особенности данных, которые вы обнаружили и на которые стоит обратить внимание при предобработке. Например, вы можете проверить названия столбцов: все ли названия отражают содержимое данных и прописаны в удобном для работы виде.

Выгрузив данные сразу видно что в них присутствуют пропуски,т.к. в строке(RangeIndex: 16956 entries, 0 to 16955) разное количество строк что говорит о наличии пропусков.Наибольшее количество пропусков встречается в столбце (Critic Score),а так же в столбцах (User Score и Rating).В некоторых столбцах неверные типы данных (object),придётся исправить.Заметно так же что встречаются названия столбцов в разных регистрах. </font>

---

## 2.  Проверка ошибок в данных и их предобработка<a id='pr'></a>


### 2.1. Названия, или метки, столбцов датафрейма

- Выведите на экран названия всех столбцов датафрейма и проверьте их стиль написания.
- Приведите все столбцы к стилю snake case. Названия должны быть в нижнем регистре, а вместо пробелов — подчёркивания.

In [2]:

display(f"Исходные названия столбцов:")
display(df.columns.tolist())
df.columns = (
 df.columns.str.lower()
 .str.replace(' ', '_')
 .str.replace('(', '')
 .str.replace(')', '')
)

display(f"Новые названия столбцов (стиль snake_case):")
display(df.columns.tolist())


'Исходные названия столбцов:'

['Name',
 'Platform',
 'Year of Release',
 'Genre',
 'NA sales',
 'EU sales',
 'JP sales',
 'Other sales',
 'Critic Score',
 'User Score',
 'Rating']

/tmp/ipykernel_114/196139304.py:4: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will*not* be treated as literal strings when regex=True.
  df.columns.str.lower()


'Новые названия столбцов (стиль snake_case):'

['name',
 'platform',
 'year_of_release',
 'genre',
 'na_sales',
 'eu_sales',
 'jp_sales',
 'other_sales',
 'critic_score',
 'user_score',
 'rating']

### 2.2. Типы данных

- Если встречаются некорректные типы данных, предположите их причины.
- При необходимости проведите преобразование типов данных. Помните, что столбцы с числовыми данными и пропусками нельзя преобразовать к типу `int64`. Сначала вам понадобится обработать пропуски, а затем преобразовать типы данных.

In [3]:
df['year_of_release'] = pd.to_numeric(df['year_of_release'], errors='coerce').dropna().astype('float64')
df['eu_sales'] = pd.to_numeric(df['eu_sales'], errors='coerce').dropna().astype('float64')
df['jp_sales'] = pd.to_numeric(df['jp_sales'], errors='coerce').dropna().astype('float64')
df['user_score'] = pd.to_numeric(df['user_score'], errors='coerce').dropna().astype('float64')
display (df.head(), df.info())


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16956 entries, 0 to 16955
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   name             16954 non-null  object 
 1   platform         16956 non-null  object 
 2   year_of_release  16681 non-null  float64
 3   genre            16954 non-null  object 
 4   na_sales         16956 non-null  float64
 5   eu_sales         16950 non-null  float64
 6   jp_sales         16952 non-null  float64
 7   other_sales      16956 non-null  float64
 8   critic_score     8242 non-null   float64
 9   user_score       7688 non-null   float64
 10  rating           10085 non-null  object 
dtypes: float64(7), object(4)
memory usage: 1.4+ MB


name platform  year_of_release         genre  na_sales  \
0                Wii Sports      Wii           2006.0        Sports     41.36   
1         Super Mario Bros.      NES           1985.0      Platform     29.08   
2            Mario Kart Wii      Wii           2008.0        Racing     15.68   
3         Wii Sports Resort      Wii           2009.0        Sports     15.61   
4  Pokemon Red/Pokemon Blue       GB           1996.0  Role-Playing     11.27   

   eu_sales  jp_sales  other_sales  critic_score  user_score rating  
0     28.96      3.77         8.45          76.0         8.0      E  
1      3.58      6.81         0.77           NaN         NaN    NaN  
2     12.76      3.79         3.29          82.0         8.3      E  
3     10.93      3.28         2.95          80.0         8.0      E  
4      8.89     10.22         1.00           NaN         NaN    NaN

None

- В числовых столбцах могут встретиться строковые значения, например `unknown` или другие. Приводите такие столбцы к числовому типу данных, заменив строковые значения на пропуски.

*Все столбцы приведены к соответсвующим типам*

### 2.3. Наличие пропусков в данных

- Посчитайте количество пропусков в каждом столбце в абсолютных и относительных значениях.


In [4]:
display (f"Количество пропусков в абсолютных значениях: {df.isna().sum()}")
display (f"Количество пропусков в относительных значениях: {df.isna().sum() / len(df)*100}")

'Количество пропусков в абсолютных значениях: name                  2\nplatform              0\nyear_of_release     275\ngenre                 2\nna_sales              0\neu_sales              6\njp_sales              4\nother_sales           0\ncritic_score       8714\nuser_score         9268\nrating             6871\ndtype: int64'

'Количество пропусков в относительных значениях: name                0.011795\nplatform            0.000000\nyear_of_release     1.621845\ngenre               0.011795\nna_sales            0.000000\neu_sales            0.035386\njp_sales            0.023590\nother_sales         0.000000\ncritic_score       51.391838\nuser_score         54.659118\nrating             40.522529\ndtype: float64'

- Изучите данные с пропущенными значениями. Напишите промежуточный вывод: для каких столбцов характерны пропуски и сколько их. Предположите, почему пропуски могли возникнуть. Укажите, какие действия с этими данными можно сделать и почему.


Для разных столбцов пропуски имеют различный характер:


name, genre, eu_sales и jp_sales: небольшое количество пропусков. Возможно забыли зафиксировать при загрузке данных;


year_of_release: 275 пропусков (1,6%). Пропуски в пределах допустимых 2%, поэтому думаю,что возможно ошибка произошла во время загрузки данных;


critic_score, user_score, rating: от 40 до 55% данных пропущено. Предполагаю, что недостаточное количество рейтинга и оценок пользователей связано с продажами, что в свою очередь связано с популярностью игры.


Продажи высокие (игра популярная)- есть оценка и рейтинг, продажи низкие (игра непопулярная) -нет оценки и рейтинга.


Принимается поступить так:


- пропуски в name: удалить (не повлияют на анализ);
- пропуски в genre: удалить (не повлияют на анализ);
- пропуски в eu_sales: удалить (практически не повлияют на анализ);
- пропуски в jp_sales: удалить (практически не повлияют на анализ);
- пропуски в year_of_release: удалить (маловероятно, что повлияют на анализ);
- пропуски в critic_score: ничего не делать;
- пропуски в user_score: ничего не делать;
- пропуски в rating: ничего не делать (не участвуют в анализе);

- Обработайте пропущенные значения. Для каждого случая вы можете выбрать оптимальный, на ваш взгляд, вариант: заменить на определённое значение, оставить как есть или удалить.
- Если вы решите заменить пропуски на значение-индикатор, то убедитесь, что предложенное значение не может быть использовано в данных.
- Если вы нашли пропуски в данных с количеством проданных копий игры в том или ином регионе, их можно заменить на среднее значение в зависимости от названия платформы и года выхода игры.

In [5]:
df = df.dropna(subset=['name', 'genre','eu_sales','jp_sales','year_of_release'])
display (df.head(),df.tail())



name platform  year_of_release         genre  na_sales  \
0                Wii Sports      Wii           2006.0        Sports     41.36   
1         Super Mario Bros.      NES           1985.0      Platform     29.08   
2            Mario Kart Wii      Wii           2008.0        Racing     15.68   
3         Wii Sports Resort      Wii           2009.0        Sports     15.61   
4  Pokemon Red/Pokemon Blue       GB           1996.0  Role-Playing     11.27   

   eu_sales  jp_sales  other_sales  critic_score  user_score rating  
0     28.96      3.77         8.45          76.0         8.0      E  
1      3.58      6.81         0.77           NaN         NaN    NaN  
2     12.76      3.79         3.29          82.0         8.3      E  
3     10.93      3.28         2.95          80.0         8.0      E  
4      8.89     10.22         1.00           NaN         NaN    NaN

name platform  year_of_release       genre  \
16951  Samurai Warriors: Sanada Maru      PS3           2016.0      Action   
16952               LMA Manager 2007     X360           2006.0      Sports   
16953        Haitaka no Psychedelica      PSV           2016.0   Adventure   
16954               Spirits & Spells      GBA           2003.0    Platform   
16955            Winning Post 8 2016      PSV           2016.0  Simulation   

       na_sales  eu_sales  jp_sales  other_sales  critic_score  user_score  \
16951      0.00      0.00      0.01          0.0           NaN         NaN   
16952      0.00      0.01      0.00          0.0           NaN         NaN   
16953      0.00      0.00      0.01          0.0           NaN         NaN   
16954      0.01      0.00      0.00          0.0           NaN         NaN   
16955      0.00      0.00      0.01          0.0           NaN         NaN   

      rating  
16951    NaN  
16952    NaN  
16953    NaN  
16954    NaN  
16955    NaN

*Произвёл обработку пропусков*


### 2.4. Явные и неявные дубликаты в данных

- Изучите уникальные значения в категориальных данных, например с названиями жанра игры, платформы, рейтинга и года выпуска. Проверьте, встречаются ли среди данных неявные дубликаты, связанные с опечатками или разным способом написания.
- При необходимости проведите нормализацию данных с текстовыми значениями. Названия или жанры игр можно привести к нижнему регистру, а названия рейтинга — к верхнему.

In [6]:
initial_row_count = df.shape[0]
df_sorted = df.sort_values(by=list(df.columns))
unique_df = df.nunique()

df_cleaned = df.drop_duplicates()
duplicates_count = initial_row_count - df_cleaned.shape[0]
final_row_count = df_cleaned.shape[0] 

unique_name = df['name'].unique()
unique_genre = df['genre'].unique()
unique_platform = df['platform'].unique()
unique_year_of_release = df['year_of_release'].unique()

display(f"Перечень уникальных названий: {unique_name}")
display(f"Перечень уникальных жанров: {unique_genre}")
display(f"Перечень уникальных платформ: {unique_platform}")
display(f"Перечень уникальных годов выпуска: {unique_year_of_release}")

display(f"Количество строк до удаления дубликатов: {initial_row_count}")
display(f"Количество найденных дубликатов: {unique_df}")
display(f"Количество строк после удаления дубликатов : {final_row_count}")
display(f"Количество удалённых строк в абсолютном значении: {initial_row_count-final_row_count}")
display(f"Процент удаленных данных: {100*(initial_row_count-final_row_count)/initial_row_count:.2f}%")

"Перечень уникальных названий: ['Wii Sports' 'Super Mario Bros.' 'Mario Kart Wii' ...\n 'Woody Woodpecker in Crazy Castle 5' 'LMA Manager 2007'\n 'Haitaka no Psychedelica']"

"Перечень уникальных жанров: ['Sports' 'Platform' 'Racing' 'Role-Playing' 'Puzzle' 'Misc' 'Shooter'\n 'Simulation' 'Action' 'Fighting' 'Adventure' 'Strategy' 'MISC'\n 'ROLE-PLAYING' 'RACING' 'ACTION' 'SHOOTER' 'FIGHTING' 'SPORTS' 'PLATFORM'\n 'ADVENTURE' 'SIMULATION' 'PUZZLE' 'STRATEGY']"

"Перечень уникальных платформ: ['Wii' 'NES' 'GB' 'DS' 'X360' 'PS3' 'PS2' 'SNES' 'GBA' 'PS4' '3DS' 'N64'\n 'PS' 'XB' 'PC' '2600' 'PSP' 'XOne' 'WiiU' 'GC' 'GEN' 'DC' 'PSV' 'SAT'\n 'SCD' 'WS' 'NG' 'TG16' '3DO' 'GG' 'PCFX']"

'Перечень уникальных годов выпуска: [2006. 1985. 2008. 2009. 1996. 1989. 1984. 2005. 1999. 2007. 2010. 2013.\n 2004. 1990. 1988. 2002. 2001. 2011. 1998. 2015. 2012. 2014. 1992. 1997.\n 1993. 1994. 1982. 2016. 2003. 1986. 2000. 1995. 1991. 1981. 1987. 1980.\n 1983.]'

'Количество строк до удаления дубликатов: 16669'

'Количество найденных дубликатов: name               11422\nplatform              31\nyear_of_release       37\ngenre                 24\nna_sales             401\neu_sales             307\njp_sales             244\nother_sales          155\ncritic_score          81\nuser_score            95\nrating                 8\ndtype: int64'

'Количество строк после удаления дубликатов : 16490'

'Количество удалённых строк в абсолютном значении: 179'

'Процент удаленных данных: 1.07%'

- После того как нормализуете данные и устраните неявные дубликаты, проверьте наличие явных дубликатов в данных.

*Найдено 179 дубликатов*

- Напишите промежуточный вывод: укажите количество найденных дубликатов и действия по их обработке.

Было принято решение удалить дубликаты потому что они препятствовали обьективному анализу.Количество дубликатов как было написанно выше было 179.Была произведена сортировка по столбцам,приведение к необходимому регистру,а так же замена данных.После проведения выше перечисленных мероприятий был получен необходимый для дальнейшего анализа датафрейм.

- В процессе подготовки данных вы могли что-либо удалять, например строки с пропусками или ошибками, дубликаты и прочее. В этом случае посчитайте количество удалённых строк в абсолютном и относительном значениях.

Найдено 179 дубликата, они же - удаленные строки в абсолютном количестве. В относительном значении - 1.07%.

- После проведения предобработки данных напишите общий промежуточный вывод.

Данные готовы для дальнейшего анализа.Все необходимые данные(такие как:год,названия игр,платформы и жанры) присутсвуют в нашем датафрейме.

---

## 3. Фильтрация данных<a id='fd'></a>

Коллеги хотят изучить историю продаж игр в начале XXI века, и их интересует период с 2000 по 2013 год включительно. Отберите данные по этому показателю. Сохраните новый срез данных в отдельном датафрейме, например `df_actual`.

In [7]:
df_actual = df[df['year_of_release'].between(2000, 2013)].copy()
display(f"Всего записей в исходных данных: {len(df)}")
display(f"Отобрано записей за 2000-2013 гг.: {len(df_actual)}")
display(f"Распределение по годам:")
display(df_actual['year_of_release'].value_counts().sort_index())

'Всего записей в исходных данных: 16669'

'Отобрано записей за 2000-2013 гг.: 12971'

'Распределение по годам:'

2000.0     357
2001.0     491
2002.0     839
2003.0     789
2004.0     769
2005.0     950
2006.0    1019
2007.0    1215
2008.0    1443
2009.0    1449
2010.0    1279
2011.0    1149
2012.0     670
2013.0     552
Name: year_of_release, dtype: int64

---

## 4. Категоризация данных<a id='cd'></a>
    
Проведите категоризацию данных:
- Разделите все игры по оценкам пользователей и выделите такие категории: высокая оценка (от 8 до 10 включительно), средняя оценка (от 3 до 8, не включая правую границу интервала) и низкая оценка (от 0 до 3, не включая правую границу интервала).

In [8]:
df_actual['user_score'] = pd.cut(df_actual['user_score'],
                               bins=[0, 3, 8, 10],
                               labels=["Низкая оценка", "Средняя оценка", "Высокая оценка"],
                               right=False)
display(df_actual)

name platform  \
0                                            Wii Sports      Wii   
2                                        Mario Kart Wii      Wii   
3                                     Wii Sports Resort      Wii   
6                                 New Super Mario Bros.       DS   
7                                              Wii Play      Wii   
...                                                 ...      ...   
16947                     Men in Black II: Alien Escape       GC   
16949                Woody Woodpecker in Crazy Castle 5      GBA   
16950  SCORE International Baja 1000: The Official Game      PS2   
16952                                  LMA Manager 2007     X360   
16954                                  Spirits & Spells      GBA   

       year_of_release     genre  na_sales  eu_sales  jp_sales  other_sales  \
0               2006.0    Sports     41.36     28.96      3.77         8.45   
2               2008.0    Racing     15.68     12.76      3.79         3.29   
3               2009.0    Sports     15.61     10.93      3.28         2.95   
6               2006.0  Platform     11.28      9.14      6.50         2.88   
7               2006.0      Misc     13.96      9.18      2.93         2.84   
...                ...       ...       ...       ...       ...          ...   
16947           2003.0   Shooter      0.01      0.00      0.00         0.00   
16949           2002.0  Platform      0.01      0.00      0.00         0.00   
16950           2008.0    Racing      0.00      0.00      0.00         0.00   
16952           2006.0    Sports      0.00      0.01      0.00         0.00   
16954           2003.0  Platform      0.01      0.00      0.00         0.00   

       critic_score      user_score rating  
0              76.0  Высокая оценка      E  
2              82.0  Высокая оценка      E  
3              80.0  Высокая оценка      E  
6              89.0  Высокая оценка      E  
7              58.0  Средняя оценка      E  
...             ...             ...    ...  
16947           NaN             NaN      T  
16949           NaN             NaN    NaN  
16950           NaN             NaN    NaN  
16952           NaN             NaN    NaN  
16954           NaN             NaN    NaN  

[12971 rows x 11 columns]

- Разделите все игры по оценкам критиков и выделите такие категории: высокая оценка (от 80 до 100 включительно), средняя оценка (от 30 до 80, не включая правую границу интервала) и низкая оценка (от 0 до 30, не включая правую границу интервала).

In [9]:
df_actual['critic_score'] = pd.cut(df_actual['critic_score'],
                                bins=[0, 30, 80, 100],
                                labels=["Низкая оценка", "Средняя оценка", "Высокая оценка"],
                                 right=False)
display(df_actual)

name platform  \
0                                            Wii Sports      Wii   
2                                        Mario Kart Wii      Wii   
3                                     Wii Sports Resort      Wii   
6                                 New Super Mario Bros.       DS   
7                                              Wii Play      Wii   
...                                                 ...      ...   
16947                     Men in Black II: Alien Escape       GC   
16949                Woody Woodpecker in Crazy Castle 5      GBA   
16950  SCORE International Baja 1000: The Official Game      PS2   
16952                                  LMA Manager 2007     X360   
16954                                  Spirits & Spells      GBA   

       year_of_release     genre  na_sales  eu_sales  jp_sales  other_sales  \
0               2006.0    Sports     41.36     28.96      3.77         8.45   
2               2008.0    Racing     15.68     12.76      3.79         3.29   
3               2009.0    Sports     15.61     10.93      3.28         2.95   
6               2006.0  Platform     11.28      9.14      6.50         2.88   
7               2006.0      Misc     13.96      9.18      2.93         2.84   
...                ...       ...       ...       ...       ...          ...   
16947           2003.0   Shooter      0.01      0.00      0.00         0.00   
16949           2002.0  Platform      0.01      0.00      0.00         0.00   
16950           2008.0    Racing      0.00      0.00      0.00         0.00   
16952           2006.0    Sports      0.00      0.01      0.00         0.00   
16954           2003.0  Platform      0.01      0.00      0.00         0.00   

         critic_score      user_score rating  
0      Средняя оценка  Высокая оценка      E  
2      Высокая оценка  Высокая оценка      E  
3      Высокая оценка  Высокая оценка      E  
6      Высокая оценка  Высокая оценка      E  
7      Средняя оценка  Средняя оценка      E  
...               ...             ...    ...  
16947             NaN             NaN      T  
16949             NaN             NaN    NaN  
16950             NaN             NaN    NaN  
16952             NaN             NaN    NaN  
16954             NaN             NaN    NaN  

[12971 rows x 11 columns]

- После категоризации данных проверьте результат: сгруппируйте данные по выделенным категориям и посчитайте количество игр в каждой категории.

In [10]:
user_score_count = df_actual['user_score'].value_counts().sort_index()
critic_score_count = df_actual['critic_score'].value_counts().sort_index()

display (f" Распределение игр по категориям user_score: {user_score_count}")
display (f" Распределение игр по категориям critic_score: {critic_score_count}")

' Распределение игр по категориям user_score: Низкая оценка      117\nСредняя оценка    4145\nВысокая оценка    2303\nName: user_score, dtype: int64'

' Распределение игр по категориям critic_score: Низкая оценка       55\nСредняя оценка    5499\nВысокая оценка    1706\nName: critic_score, dtype: int64'

- Выделите топ-7 платформ по количеству игр, выпущенных за весь актуальный период.

In [11]:
top_platform = df_actual['platform'].value_counts().head(7)
display (top_platform)

PS2     2153
DS      2143
Wii     1294
PSP     1198
X360    1135
PS3     1107
GBA      825
Name: platform, dtype: int64

---

## 5. Итоговый вывод<a id='end'></a>

В конце напишите основной вывод и отразите, какую работу проделали. Не забудьте указать описание среза данных и новых полей, которые добавили в исходный датасет.

Быд дан срез данных которые были собраны из открытых источников.В них есть информация о продажах игр, сделанных в разных жанрах и выпущенных на разных платформах, а также пользовательские и экспертные оценки игр. По ним создан датафрейм, проведена предобработка - нормализация значений, переименование и замена позиций, приведение к единому стилю, поиск и удалений дубликатов и ненужных данных для анализа.

По заданию нужно было произвести анализ среза данных с 2000г. по 2013г.,а так же разделить данные на 2 категории:
- Оценка игроков
- Оценка критиков

В конце было выделено топ-7 самых популярных платформ.После группировки и категоризации пользовательских оценок видно, что в подавляющем большинстве игроки довольны играми, в основом игры получили среднюю и высокие оценки.После групировки и категоризации заметно что оценки игроков и оценки критиков которые они ставят играм примерно одинаковое.

- Низкая оценка (0-30 баллов): 55
- Средняя оценка (30-80 баллов: 5499
- Высокая оценка (80-100 баллов): 1706.

Как и было сказанно выше в конце проекта было представленна таблица отражающяя топ-7 платформ.Лидером данного топа оказалась платформа PS2:

- PS2: 2153 игр

На втором и третьем месте расположились платформы DS и Wii:

- DS: 2143 игр
- Wii: 1294 игр

Так же в топе смогли отличится такие платформы как:

- PSP: 1198 игр
- X360: 1135 игр
- PS3: 1107 игр
- GBA: 825 игр